Project/Assets/_Scripts/_new_Scripts/Weapons/RaycastGunViewModel.cs

```diff
@@ -2,123 +2,141 @@
+        public override void StopAttack()
         {
-            StartCoroutine(ExecuteReload());
+            model.ShotChargeTimer = 0f;
         }
-    }
 
-    /// <summary>
-    /// For now Instantiates ray for specified time and destroys it after
-    /// </summary>
-    private IEnumerator ShotEffect()
-    {
-        activeRay = Instantiate(LineRendererPrefab).GetComponent<LineRenderer>();
-        LineRenderer newRay = activeRay;
-        yield return new WaitForSeconds(model.ShotDuration.Value);
-        Destroy(newRay.gameObject);
-    }
+        /// <summary>
+        /// Function called on pressing the ReloadButton
+        /// </summary>
+        public override void Reload()
+        {
+            if (model.BulletsInMagazine < model.MagazineSize && model.Reloading == false)
+            {
+                StartCoroutine(ExecuteReload());
+            }
+        }
 
-    private IEnumerator ExecuteReload()
-    {
-        model.Reloading.Value = true;
-        yield return new WaitForSeconds(model.ReloadTime.Value);
-        model.Reloading.Value = false;
-        model.BulletsInMagazine.Value = model.MagazineSize.Value;
-    }
+        /// <summary>
+        /// For now Instantiates ray for specified time and destroys it after
+        /// </summary>
+        private IEnumerator ShotEffect()
+        {
+            activeRay = Instantiate(LineRendererPrefab).GetComponent<LineRenderer>();
+            LineRenderer newRay = activeRay;
+            yield return new WaitForSeconds(model.ShotDuration);
+            Destroy(newRay.gameObject);
+        }
 
-    private void ExecuteAttack()
-    {
-        model.BulletsInMagazine.Value--;
+        private IEnumerator ExecuteReload()
+        {
+            model.Reloading = true;
+            yield return new WaitForSeconds(model.ReloadTime);
+            model.Reloading = false;
+            model.BulletsInMagazine = model.MagazineSize;
+        }
 
-        for (int i = 0; i < model.BulletCount.Value; ++i)
+        private void ExecuteAttack()
         {
-            model.PenetrationDamage.Value = model.Damage.Value;
-            model.PenetrationLeft.Value = model.BulletPenetration.Value;
+            model.BulletsInMagazine--;
 
-            StartCoroutine(ShotEffect());
+            for (int i = 0; i < model.BulletCount; ++i)
+            {
+                model.PenetrationDamage = model.Damage;
+                model.PenetrationLeft = model.BulletPenetration;
 
-            Vector3 direction = ProjectileSpawn.right;
-            direction.y += Random.Range(-model.BulletSpread.Value, model.BulletSpread.Value); // Creating bullet spread
+                StartCoroutine(ShotEffect());
 
-            Vector3 rayOrigin = ProjectileSpawn.position;
-            activeRay.SetPosition(0, rayOrigin);
+                Vector3 direction = ProjectileSpawn.right;
+                direction.y += Random.Range(-model.BulletSpread, model.BulletSpread); // Creating bullet spread
```

Comment:
To bym owinął w jakąś właściwość, która zwraca floata.

Reply:
Poprawione, nawet przyjemnie wygląda